### PR TITLE
Remove unused functions, cimports

### PR DESCRIPTION
**Pull request by jbrockmendel**
Miscellaneous cleanup, use `util.is_array` instead of `isinstance`, that kind of thing.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/reduction.pyx [hunk 1]
```diff
@@ -24,9 +24,9 @@ is_numpy_prior_1_6_2 = LooseVersion(np.__version__) < '1.6.2'
 
 cdef _get_result_array(object obj, Py_ssize_t size, Py_ssize_t cnt):
 
-    if isinstance(obj, np.ndarray) \
-            or isinstance(obj, list) and len(obj) == cnt \
-            or getattr(obj, 'shape', None) == (cnt,):
+    if (util.is_array(obj) or
+            isinstance(obj, list) and len(obj) == cnt or
```
Comment (jreback) [inline pandas/_libs/reduction.pyx]: could do PyListCheck here

Comment (jreback) [reply pandas/_libs/reduction.pyx]: or change the other below, I find this slightly non-idiomatic

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/reduction.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/reduction.pyx]: It looks like cython automatically converts to `PyList_Check`.

Agreed on a lot of the idiomatic point.  There's a lot of built-up debris in _libs.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**